File: C/Algorithms/1979.c
int findGCD(int* nums, int numsSize){
    int min = 1001, max = 0;
    for (int i = 0; i < numsSize; i++){
        int num = nums[i];
        if (num > max)
            max = num;
        if (num < min)
            min = num;
    }
    
    while (min != 0){
        int temp = min;
        min = max % min;
        max = temp;
    }
    return max;
}

File: Rust/Algorithms/1979.rs
impl Solution {
    pub fn find_gcd(nums: Vec<i32>) -> i32 {
        let (mut min, mut max) = (1001, 0);
        for num in nums {
            if num > max {
                max = num;
            }
            if num < min {
                min = num;
            }
        }
        
        while min != 0 {
            // ok in Rust 1.59.
            // (min, max) = (max % min, min);
            let temp = min;
            min = max % min;
            max = temp;
        }
        max
    }
}